core/crypto/sm3.c

```diff
@@ -55,7 +55,7 @@ static void __maybe_unused sm3_process(struct sm3_context *ctx,
 {
 	uint32_t SS1, SS2, TT1, TT2, W[68], W1[64];
 	uint32_t A, B, C, D, E, F, G, H;
-	uint32_t T[64];
+	uint64_t T[64];
```

Comment:
I'm not sure this will give the desired result. Please fix the `SHL()` macro instead.

Reply:
I will proceed as I mentionned earlier then https://github.com/OP-TEE/optee_os/pull/6591#issuecomment-1885208252